client/src/shared/hooks/useShowUserList.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import API from '@/src/apis';
+import { useEffect, useState } from 'react';
+import { getUserToken } from '@/src/shared/utils';
+import { useRouter } from 'next/router';
+import type { UserList } from '@/src/shared/utils';
+
+const useShowUserList = (): any => {
+  const router = useRouter();
+  const params = router.query;
+  const [listOfUser, setListOfUser] = useState<UserList[]>([]);
+  const numberOfUsers = listOfUser.length;
+  const [limiter, setLimiter] = useState(10);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [showPerPage, setShowPerPage] = useState<UserList[]>([]);
+  const [startingIndex, setStartingIndex] = useState(0);
+  const [lastIndex, setLastIndex] = useState(0);
+
+  const handleChangePageEvent = (page: number): void => {
+    async function fetchdata (): Promise<void> {
+      try {
+        const token = getUserToken();
+        const headers = { Authorization: `Token ${token}` };
+        const response = await API.get(
+          `user/${params.company_id}?page_size=${limiter}&page=${page}`,
+          {
+            headers
+          }
+        );
+        setShowPerPage(response.data.user);
+      } catch (error) {
+        console.error(error);
+      }
```

Comment:
No need to redefine the variable token in here since you are already imported the API from apis folder. Where the token is already fetched in that function
```suggestion
      try {
        const response = await API.get(
          `user/${params.company_id}?page_size=${limiter}&page=${page}`
        );
        setShowPerPage(response.data.user);
      } catch (error) {
        console.error(error);
      }
```